pkg/dao/workflow.go

```diff
@@ -394,16 +395,18 @@ func OverwriteWorkflowVariables(vars map[string]string, config types.WorkflowVar
 		if !configKeys.Has(k) {
 			return nil, fmt.Errorf("invalid variables: %s", k)
 		}
+
+		mergedVars[k] = vars[k]
 	}
 
 	for i := range config {
 		name, value := config[i].Name, config[i].Value
-		if _, ok := vars[name]; !ok {
-			vars[name] = value
+		if _, ok := mergedVars[name]; !ok {
+			mergedVars[name] = value
 		}
 	}
```

Comment:
Can we set these in L387-L392 so that we only loop config once?

Reply:
updated